src/components/link-preview/link-preview.pw.tsx

```diff
@@ -0,0 +1,266 @@
+import {
+  linkPreview,
+  linkPreviewCloseIcon,
+  linkPreviewAs,
+} from "../../../playwright/components/link-preview";
+import {
+  checkGoldenOutline,
+  checkAccessibility,
+} from "../../../playwright/support/helper";
+import { CHARACTERS } from "../../../playwright/support/constants";
+import { HooksConfig } from "../../../playwright";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const testCypress = "test-cypress";
+const urlProp = "./carbon-by-sage-logo.png";
+const keysToTrigger = ["Space", "Enter"];
+
+test.describe("when focused", () => {
+  test("should have the expected styling when the focusRedesignOptOut is false", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<LinkPreviewComponentTest aria-label={CHARACTERS.STANDARD} />);
+
+    const previewElement = linkPreview(page);
+    await previewElement.focus();
+
+    await expect(previewElement).toHaveCSS(
+      "box-shadow",
+      "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+    );
+  });
+
+  test("should have the expected styling when the focusRedesignOptOut is true", async ({
+    mount,
+    page,
+  }) => {
+    await mount<HooksConfig>(
+      <LinkPreviewComponentTest aria-label={CHARACTERS.STANDARD} />,
+      {
+        hooksConfig: { focusRedesignOptOut: true },
+      }
+    );
+
+    const previewElement = linkPreview(page);
+    await previewElement.focus();
+
+    await expect(previewElement).toHaveCSS(
+      "outline",
+      "rgb(255, 188, 25) solid 2px"
+    );
+  });
+
+  test.describe("check props for Link Preview component", () => {
+    (["div", "a"] as LinkPreviewProps["as"][]).forEach((as) => {
+      test(`should render Link Preview as prop using ${as}`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<LinkPreviewComponentTest as={as} />);
+
+        await expect(linkPreviewAs(page, as)).toBeVisible();
+      });
+    });
+
+    testData.forEach((title) => {
+      test(`should render Link Preview title prop using ${title} as special character`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<LinkPreviewComponentTest title={title} />);
+
+        await expect(page.getByText(title)).toBeAttached();
+      });
+    });
+
+    testData.forEach((description) => {
+      test(`should render Link Preview description prop using ${description} as special character`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(<LinkPreviewComponentTest description={description} />);
+
+        await expect(page.getByText(description)).toBeAttached();
+      });
+    });
+
+    test("should render Link Preview with isLoading prop", async ({
+      mount,
+      page,
+    }) => {
+      await mount(<LinkPreviewComponentTest isLoading />);
+
+      const previewComponent = page.locator(`[data-component="preview"]`);
+
+      const count = await previewComponent.count();
+
+      expect(count).toBe(4);
```

Comment:
I've observed that attribute checking has been done in cypress but missed here  "("have.attr", "data-component", "preview")"

Reply:
Hi @divyajindel, this is because the line for testing "have.attr" becomes a bit useless, as I already have the line99, which states that "page.locator(`[data-component="preview"]`);", we used this way to locate [data-component="preview"], which we should definitely have attribute "preview" as our "data-component"